conjure-java-core/src/main/java/com/palantir/conjure/java/types/BeanBuilderGenerator.java

```diff
@@ -336,15 +336,16 @@ private CodeBlock typeAwareAssignment(EnrichedField enriched, Type type, boolean
                     .build();
         } else if (type.accept(TypeVisitor.IS_OPTIONAL)) {
             OptionalType optionalType = type.accept(TypeVisitor.OPTIONAL);
+            String parameterName = enriched.poetSpec().name;
             CodeBlock nullCheckedValue = Expressions.requireNonNull(
                     spec.name, enriched.fieldName().get() + " cannot be null");
 
             if (isWidenableContainedType(optionalType.getItemType())) {
                 // covariant optionals need to be narrowed to invariant type before assignment
                 Type innerType = optionalType.getItemType();
                 return CodeBlock.builder()
-                        .addStatement("this.$1N = ($3T<$4T>) $2L",
-                                spec.name, nullCheckedValue, Optional.class, typeMapper.getClassName(innerType).box())
+                        .addStatement("this.$1N = $2L.map(_$3L -> ($4T) _$3L)",
+                                spec.name, nullCheckedValue, parameterName, typeMapper.getClassName(innerType).box())
```

Comment:
I think i'd use `_value` as a fixed name rather than `parameterName`.

Reply:
used `Function.identity()` generic to instead capture the covariant type (thanks for the offline recommendation)